### ISSUE TITLE
Default user user roles to include Publisher and Editor

### ISSUE THREAD

**Issue by clarkepaul** [silverstripe/silverstripe-framework#8597]
## Description

With the default setup we provide an Admin group as the only group defined in the CMS (it has permission applied directly to the group rather than a role being setup). The initial Admin will often invite other team members to the CMS admin area within the same Admin group as it is the only thing which is setup. I think that it's expected that the developers will define the groups and roles prior to handing over the site to Admins but this would be often overlooked by the developers, and often not further defined by Admins.

This could be seen as a security issue if everyone who has admin area access is an Admin, when they they haven't had the appropriate training or understanding of the implications of what they are doing.

The permission within the CMS are actually very complicated in the way they are configured, and its very hard to figure out what is overriding what permission ([issue covering overriding permissions](https://github.com/silverstripe/silverstripe-framework/issues/4861)).

The user help explains how to use Roles but this doesn't align to the default way the CMS is configured where permissions are applied directly to groups (separate issue to cover this). Applying permissions directly to groups can make sense if there are only a few groups and you want to keep things simple but if there are more than a few groups it does make sense to use the roles properly to allow for expansion. If they were predefined people might have a better indication of how to use them.

 Main areas which have permissions:
- Site settings
- Group permissions
- Role permissions
- Content settings/permissions
- Access permissions for areas of the CMS
- ... and probably more 

## Steps to Reproduce
- New installation of CMS
- Add new person to Security/Members
- Add new person to Group
- Only Admin group is setup

## Proposed solution

We provide some good default Groups with Roles predefined so that admins can then alter and tailor to their specific needs. Not only will this make it easier for developers but will also provide Admins a good example of how things can be set up—a better example than trying to explain it in user help.

Suggested groups/roles:
Admins -> Administrator Role
Publishers -> Content Publisher
Editors -> Content Editor

**Comment by robbieaverill** [comment]
As a side note, I'm thinking about raising an RFC to rewrite the ACL system we have in SilverStripe, maybe for SS5. The issue you mentioned around not being able to override core permissions is an uncommon but annoying pain point and a limitation of our current architecture. A more robust ACL system would allow core, modules and user code to more control allow and disallow for permission codes in a more flexible level.

**Comment by sminnee** [comment]
Adding Publisher and Editor as default roles could be a good idea. For editor this would mean that they could edit draft but not do any publication themselves. The publisher would need to log in separately to press publish.

Anything involving request publication, notification, is going to lead us down the path of accidentally reimplementing workflow, so if you feel that this are necessary to make the editor/publisher roles "finished" I suggest you hold fire.

If you install the blog module, I would suggest that we have Blog Editor and Blog Publisher that give rights for a single default BlogPage and its descendants. Notably this would affect some of our amin reipes.

**Comment by clarkepaul** [comment]
Yeah nothing along the lines of approval :) thats a bigger thing.
